docs/build/guides/conventions/working-with-contract-specs.mdx

```diff
@@ -0,0 +1,1264 @@
+---
+title: Work with contract specs in Java, Python, and PHP
+description: A guide to understanding and interacting with Soroban smart contracts in different programming languages.
+hide_table_of_contents: false
+---
+
+import { CodeExample } from "@site/src/components/CodeExample";
+
+## Introduction
+
+Soroban smart contracts are powerful tools for building decentralized applications on the Stellar network. To interact with these contracts effectively, it's crucial to understand their specifications and how to use them in your programming language of choice.
+
+A typical contract specification (spec) includes:
+
+1. Data types used by the contract
+2. Function definitions with their inputs and outputs
+3. Error types that the contract may return
+
+These details guide how you interact with the contract, regardless of the programming language you're using.
+
+## Prerequisites
+
+Before diving into contract interactions, ensure you have the following:
+
+- Stellar CLI ([`stellar`](../../smart-contracts/getting-started/setup.mdx#install-the-stellar-cli)) installed
+- A Soroban-compatible SDK for your programming language. View the [list of available SDKs](../../../tools/sdks/library.mdx) to find one that suits your needs
+- Access to a Stellar [RPC server](../../../learn/fundamentals/networks.mdx) (local or on a test network)
+
+For this guide, we will focus on the Java, Python, and PHP SDKs for reference, but the concepts can also be applied to other languages.
```

Comment:
Would be really nice to have `Java`, `Python`, and `PHP` as links to the respective SDK documentation.